src/v/cluster/partition_balancer_backend.cc

```diff
@@ -145,9 +145,16 @@ ss::future<> partition_balancer_backend::do_tick() {
     _last_violations = std::move(plan_data.violations);
 
     if (
-      _topic_table.has_updates_in_progress() || !plan_data.reassignments.empty()
-      || !plan_data.cancellations.empty()) {
+      _topic_table.has_updates_in_progress()
+      || plan_data.status
+           == partition_balancer_planner::status::cancellations_planned
+      || plan_data.status
+           == partition_balancer_planner::status::movement_planned) {
         _last_status = partition_balancer_status::in_progress;
+    } else if (
+      plan_data.status
+      == partition_balancer_planner::status::waiting_for_reports) {
+        _last_status = partition_balancer_status::waiting_for_reports;
     } else if (plan_data.failed_reassignments_count > 0) {
```

Comment:
would you mind cleaning this up some either with `// clang-format off` or with some pre-calculated bools to use within the conditionals?

Reply:
added `planner_status` using
now it look better